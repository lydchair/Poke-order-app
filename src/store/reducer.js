import * as actionTypes from './actions';

const initialState = {
    ingredients: {
        salad: 0,
        tofu: 0,
        tuna: 0,
        salmon: 0
    },
    totalPrice: 7.5
};

const INGREDIENT_PRICES = {
    salad: 0.5,
    tuna: 1,
    salmon: 1,
    tofu: 0.8
};

const reducer = ( state = initialState, action ) => {
    switch ( action.type ) {
        case actionTypes.ADD_INGREDIENT:
            return {
                ...state,
                ingredients: {
                    ...state.ingredients,
                    [action.ingredientName]: state.ingredients[action.ingredientName] + 1
                },
                totalPrice: state.totalPrice + INGREDIENT_PRICES[action.ingredientName]
            };
        case actionTypes.REMOVE_INGREDIENT:
            return {
                ...state,
                ingredients: {
                    ...state.ingredients,
                    [action.ingredientName]: state.ingredients[action.ingredientName] - 1
                },
                totalPrice: state.totalPrice - INGREDIENT_PRICES[action.ingredientName]
            };
        default:
            return state;
    }
};

export default reducer;