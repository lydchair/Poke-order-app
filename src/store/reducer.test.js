import React from 'react';
import renderer from 'react-test-renderer';

import reducer from './reducer';

describe('reducer ', () => {
    it('should return initial state', () => {
      expect(reducer(undefined, {})).toMatchSnapshot();
    });
  
    it('should handle ADD_INGREDIENT', () => {
      const action = {
        type: 'ADD_INGREDIENT',
        ingredient: '',
        totalPrice:'',
      };
  
      expect(reducer(undefined, action)).toMatchSnapshot();
    });
  
    it('should handle UPDATE_NAME', () => {
      const action = {
        type: 'REMOVE_INGREDIENT',
        ingredient: '',
        totalPrice:'',
      };
  
      expect(reducer(undefined, action)).toMatchSnapshot();
    });
  });