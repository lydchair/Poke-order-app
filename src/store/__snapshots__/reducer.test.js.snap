// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer  should handle ADD_INGREDIENT 1`] = `
Object {
  "ingredients": Object {
    "salad": 0,
    "salmon": 0,
    "tofu": 0,
    "tuna": 0,
    "undefined": NaN,
  },
  "totalPrice": NaN,
}
`;

exports[`reducer  should handle UPDATE_NAME 1`] = `
Object {
  "ingredients": Object {
    "salad": 0,
    "salmon": 0,
    "tofu": 0,
    "tuna": 0,
    "undefined": NaN,
  },
  "totalPrice": NaN,
}
`;

exports[`reducer  should return initial state 1`] = `
Object {
  "ingredients": Object {
    "salad": 0,
    "salmon": 0,
    "tofu": 0,
    "tuna": 0,
  },
  "totalPrice": 7.5,
}
`;
